src/dbnode/server/server.go

```diff
@@ -1165,6 +1167,100 @@ func kvWatchEncodersPerBlockLimit(
 	}()
 }
 
+func kvWatchQueryLimit(
+	store kv.Store,
+	logger *zap.Logger,
+	limits limits.QueryLimits,
+	defaultOpts limits.Options,
+) {
+	value, err := store.Get(kvconfig.QueryLimits)
+	if err == nil {
+		dynamicLimits := &kvpb.QueryLimits{}
+		err = value.Unmarshal(dynamicLimits)
+		if err == nil && dynamicLimits != nil {
```

Comment:
more of a question: how would dynamicLimits ever be nil at this point?

Reply:
Ah that's a mistake. Will fix